lib/metrics/StatsModel.js

```diff
@@ -88,10 +96,22 @@ class StatsModel extends StatsClient {
                 return cb(null, statsRes);
             }
 
-            statsRes.requests = this._zip(requests).map(arr =>
+            requests = this._zip(requests).map(arr =>
                 arr.reduce((acc, i) => acc + i));
-            statsRes['500s'] = this._zip(errors).map(arr =>
+            errors = this._zip(errors).map(arr =>
                 arr.reduce((acc, i) => acc + i));
+
+            // replace any existing values in arrays "requests" and "errors"
+            // in statsRes.requests and statsRes['500s']
+            // i.e. statsRes.requests = [0,0,0,0]
+            //      requests = [1, 2]
+            //    The splice should change:
+            //      statsRes.requests = [1,2,0,0]
+            Array.prototype.splice.apply(statsRes.requests,
```

Comment:
For clarity I think you could call the function normally and pass the remaining arguments as `...requests`.